docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,957 @@
+
+`@Step()` takes an optional `StepConfig`, which allows a number of step properties to be specified:
+
+```typescript
+export interface StepConfig {
+  retriesAllowed?: boolean; // Should failures be retried? (default true)
+  intervalSeconds?: number; // Seconds to wait before the first retry attempt (default 1).
+  maxAttempts?: number;     // Maximum number of retry attempts (default 3). If errors occur more times than this, throw an exception.
+  backoffRate?: number;     // Multiplier by which the retry interval increases after a retry attempt (default 2).
+}
+```
+
+Example:
+```typescript
+@DBOS.step()
+static async sendToServer(valueToSend: string) {
+  ...
+}
+```
+
+## Accessing Application Functions Requiring Context Arguments
+Prior versions of the DBOS SDK were based on functions that took a [`context`](./contexts.md#dboscontext) as the first argument.  It is possible to call these old-style step and transaction functions from new workflows via the `DBOS.invoke` syntax:
+
+```typescript
+DBOS.invoke<T extends ConfiguredInstance>(targetInst: T): InvokeFuncsInst<T>;
+DBOS.invoke<T extends object>(targetClass: T): InvokeFuncs<T>
+```
+
+Example:
+```typescript
+class MyClass {
+  // Older dercorator, function takes `ctx` for SQL access
+  @Transaction()
+  static async transactionFunction(ctx: TransactionContext, arg: string) {
+    //...
+  }
+}
+
+// Call with `DBOS.invoke`
+const res = DBOS.invoke(MyClass).transactionFunction('arg');
+```
+
+## Sleeping
+
+Within a DBOS workflow, waiting or sleeping should not be done using standard system functions, as these will not be skipped on workflow replay.  Instead, the "durable sleep" functions below should be used.  The wakeup time will be stored in the database when the function is first called, and if the workflow is re-executed, it will not oversleep.
+
+```typescript
+DBOS.sleepms(durationMS: number): Promise<void>
+DBOS.sleep(durationSec: number): Promise<void>
+```
+
+* DBOS.sleep: sleep for `durationSec` seconds.
+* DBOS.sleepms: sleep for `durationMS` milliseconds.
+
+These functions work in any context, and will use the system sleep if no workflow is in progress.
+
+## Interacting With Workflows
+
+### Sending And Receiving Messages
+
+`DBOS.send` and `DBOS.recv` allows the sending of messages to a specific [workflow](../tutorials/workflow-tutorial#workflow-identity).  Workflows may wait for the message to be received before proceeding.
+
+#### `DBOS.send`
+```typescript
+DBOS.send<T>(destinationID: string, message: T, topic?: string): Promise<void>
+```
+`DBOS.send()` to send a message to a workflow, identified by `destinationID`.  Messages can optionally be associated with a topic and are queued on the receiver per topic.  `DBOS.send` may be called from other workflows, or anywhere else, to pass information to a corresponding `DBOS.recv` call.
+
+For more information, see our [messages API tutorial](../tutorials/workflow-communication-tutorial#messages-api).
```

Comment:
This link is about using contexts from v1. Maybe remove this reference for now.

Reply:
I was thinking we would rewrite this, and other things that we're referring to, in the next round - up for review early next week and merged soon thereafter.  A link to the old one is better than none, for now.